MinecraftClient/Mapping/Movement.cs

```diff
@@ -129,62 +129,89 @@ public static Queue<Location> Move2Steps(Location start, Location goal, ref doub
         /// <param name="start">Start location</param>
         /// <param name="goal">Destination location</param>
         /// <param name="allowUnsafe">Allow possible but unsafe locations</param>
+        /// <param name="maxOffset">If no valid path can be found, also allow locations within specified distance of destination</param>
+        /// <param name="minOffset">Do not get closer of destination than specified distance</param>
         /// <returns>A list of locations, or null if calculation failed</returns>
-        public static Queue<Location> CalculatePath(World world, Location start, Location goal, bool allowUnsafe = false)
+        public static Queue<Location> CalculatePath(World world, Location start, Location goal, CancellationToken ct, bool allowUnsafe=false, int maxOffset=0, int minOffset=0)
         {
-            Queue<Location> result = null;
 
-            AutoTimeout.Perform(() =>
+            if (minOffset > maxOffset)
+                throw new ArgumentException("maxOffset must be >= minOffset");
```

Comment:
I'd suggest to reverse the argument exception since `minOffset` is specified after `maxOffset`.
```suggestion
            if (minOffset > maxOffset)
                throw new ArgumentException("minOffset must be lower than or equal to maxOffset", "minOffset");
```
Also changed to the constructor that explicitly states the name of the affected parameter.